### ISSUE TITLE
Twitter tests fail

### ISSUE THREAD

**Issue by mtrezza** [parse-community/Parse-SDK-JS#1917]
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/Parse-SDK-JS/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the latest versions of [Parse Server](https://github.com/parse-community/parse-server/releases) and the [Parse JS SDK](https://github.com/parse-community/Parse-SDK-JS/releases). 

### Issue Description

Twitter tests fail, see https://github.com/parse-community/Parse-SDK-JS/actions/runs/5081498916/jobs/9129960472?pr=1916

This is likely a twitter expired token issue.

**Comment by dblythy** [comment]
```
  To generate the auth data below, the Twitter app "GitHub CI Test App" has
  been created, managed by the @ParsePlatform Twitter account. In case this
  test starts to fail because the token has become invalid, generate a new
  token according to the OAuth process described in the Twitter docs[1].

  [1] https://developer.twitter.com/en/docs/authentication/oauth-1-0a/obtaining-user-access-tokens
```

**Comment by mtrezza** [comment]
The question is whether we should manually generate a new token. Every year or so we need to update this thing. Is there an alternative? Maybe we could generate the token as part of the CI? With the necessary credentials in GitHub secrets.

**Comment by dplewis** [comment]
Manually generate a new token looks like @mtrezza did it last time. The CI might not be the best option since we would still like to have local testing.

**Comment by mtrezza** [comment]
I think the CI should generate a token for every run, so we don't have to renew this manually every year or so; not sure how complex that is. For local testing people would have to use their own Twitter token. I think the current solution is not good because it allows misuse of our Twitter app, even though it's just a test app. Any rate limits or misuse would fall back on us because we publicly share the tokens.

**Comment by mtrezza** [comment]
In fact, our Twitter test account has been suspended, that's why the CI doesn't work anymore. We shouldn't share these details.

**Comment by mtrezza** [comment]
In fact, our Twitter test app has been suspended, that's why the CI doesn't work anymore. Looking into it...

**Comment by mtrezza** [comment]
I think the solution here is to mock the Twitter API as an immediate fix. 

Then, if someone wants to take it on in the future, we may use the real Twitter API, but not the was it is currently done. The token that is now hard coded in the tests needs to be requested from Twitter so that we don't have to renew this every year. The credentials needed need to be stored as GitHub secrets, unavailable to the public. That means Twitter test will fail locally, unless someone uses their own Twitter app credentials for development.

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [4.2.0-alpha.4](https://github.com/parse-community/Parse-SDK-JS/releases/tag/4.2.0-alpha.4)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [4.3.0-beta.1](https://github.com/parse-community/Parse-SDK-JS/releases/tag/4.3.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [4.3.0-alpha.1](https://github.com/parse-community/Parse-SDK-JS/releases/tag/4.3.0-alpha.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [4.3.0](https://github.com/parse-community/Parse-SDK-JS/releases/tag/4.3.0)
